ledger/src/blockstore_db.rs

```diff
@@ -1335,6 +1397,50 @@ fn get_cf_options<C: 'static + Column + ColumnName>(
     options
 }
 
+fn new_cf_descriptor_fifo<C: 'static + Column + ColumnName>(
+    max_cf_size: &u64,
+) -> ColumnFamilyDescriptor {
+    ColumnFamilyDescriptor::new(C::NAME, get_cf_options_fifo::<C>(max_cf_size))
+}
+
+/// Returns the RocksDB Column Family Options which use FIFO Compaction.
+///
+/// Note that this CF options is optimized for workloads which write-keys
+/// are mostly monotonically increasing over time.  For workloads where
+/// write-keys do not follow any order in general should use get_cf_options
+/// instead.
+///
+/// - [`max_cf_size`]: the maximum allowed column family size.  Note that
+/// rocksdb will start deleting the oldest SST file when the column family
+/// size reaches `max_cf_size` - `FIFO_WRITE_BUFFER_SIZE` to strictly
+/// maintain the size limit.
+fn get_cf_options_fifo<C: 'static + Column + ColumnName>(max_cf_size: &u64) -> Options {
+    let mut options = Options::default();
+
+    options.set_max_write_buffer_number(8);
+    options.set_write_buffer_size(FIFO_WRITE_BUFFER_SIZE as usize);
+    options.set_num_levels(1);
+    options.set_max_open_files(-1);
+
+    let mut fifo_compact_options = FifoCompactOptions::default();
+
+    // Note that the following actually specifies size trigger for deleting
+    // the oldest SST file instead of specifying the size limit as its name
+    // might suggest.  As a result, we should trigger the file deletion when
+    // the size reaches `max_cf_size - write_buffer_size` in order to correctly
+    // maintain the storage size limit.
+    fifo_compact_options.set_max_table_files_size(if *max_cf_size > FIFO_WRITE_BUFFER_SIZE {
+        *max_cf_size - FIFO_WRITE_BUFFER_SIZE
```

Comment:
Want to add a comment here.

Since `set_max_table_files_size` actually sets the deletion size trigger instead of really ensuring the size is always below the configured size, I've changed it to `max_cf_size - FIFO_WRITE_BUFFER_SIZE` to make it really always maintain the CF size below `max_cf_size`.

Reply:
prefer to use `saturating_sub` here to avoid unchecked math